src/bregman.jl

```diff
@@ -124,15 +127,29 @@ function bregman(funobj::Function, x::AbstractArray{T}, options::BregmanParams,
         # Update z variable
         @. z = z + d
         # Get λ at first iteration
-        i == 1 && (λ = abs(T(quantile(abs.(z), options.quantile))))
+        if i == 1
+            if ~isnothing(thresholdfunc)
+                λ = thresholdfunc(z)
+            elseif ~isnothing(options.lambda)
+                λ = options.lambda
+            else
+                λ = quantile(abs.(z), options.quantile)
+            end
+            λ = abs.(T.(λ))
+            sol.λ = abs.(T.(λ))
+        end
         # Save curent state
         options.spg && (gold .= g; xold .= x)
         # Update x
         x = TD'*soft_thresholding(z, λ)
 
-        obj_fun = λ * norm(z, 1) + .5 * norm(z, 2)^2
+        obj_fun = norm(λ .* z, 1) + .5 * norm(z, 2)^2
         if options.verbose > 0
-            @printf("%10d %15.5e %15.5e %15.5e %15.5e \n",i, t, obj_fun, f, λ)
+            if length(λ) == 1
+                @printf("%10d %15.5e %15.5e %15.5e %15.5e \n",i, t, obj_fun, f, λ)
+            else
+                @printf("%10d %15.5e %15.5e %15.5e %5s \n",i, t, obj_fun, f, "vector")
```

Comment:
that's not informative, choose a value like min/max/mean/ or tuple or something to give proper information